### PR TITLE
[Sage-718]- chore: remove legacy theme

### PR DESCRIPTION
**Pull request by ju-Skinner**
## Description
This is clearly a big PR. To help with reviewing, I'd suggest to either go commit by commit or pull it down locally and run. Should you prefer to do the latter, please make sure you stop all things Sage and start them back up. This will ensure that there are now cached objects and things compile as expected. 

## Screenshots
I'm electing not to include Screenshots, because I want to make sure we run locally. 

## Testing in `sage-lib`
Things to confirm:
* Docs site: http://localhost:4000
* Storybook: should now be on port 4100 again running the correct version of Sage (formerly Next)
* Style Dictionary Tokens:
  * React: packages/sage-react/lib/configs/dictionary/tokens.js
  * Rails: docs/lib/sage_rails/app/sage_tokens/

## Testing in `kajabi-products`
Separate PR will be issued for KP see [SAGE-719](https://kajabi.atlassian.net/browse/SAGE-719)

### CODE REVIEW

**Review by QuintonJason — APPROVED** [review]
Left one comment but this is good to go. `sage-lib` works and looks as expected

**Review by anechol — CHANGES_REQUESTED** [review]
Checked every Rails, React, and Sage Docs page I could find. Just one thing:

The [Grid Templates](http://localhost:4000/pages/patterns/grid_templates) page is broken due to a syntax error. Once that's done, we're good to go. Thanks for the huge effort!

**Review by anechol — APPROVED** [review]
Looking good!!

**Review by monkeypox8 — COMMENTED** [review]
This is looking great! Overall functionality is working as expected from what I'm seeing so far.

Question for the group: thoughts on keeping some aspect of the base "theme" structure in place? While `legacy`'s duplication should go, I could see a benefit in the ability to swap styles based off of CSS custom props/tokens, while keeping standard component markup/JS intact. But this being at the expense of complexity, I suppose a different approach to theming makes more sense.

There's also a few areas where the React/Storybook link is not functioning (Alert, Button, basically anywhere without a `Default` story view) but those are minor items that can be cleaned up later.

File: docs/app/views/application/_assistant.html.erb [hunk 1]
```diff
@@ -6,7 +6,7 @@
       label_text: "Use Next theme",
```
[the remark this answers is not in the document]
Comment (ju-Skinner) [reply docs/app/views/application/_assistant.html.erb:6]: Good catch, this should be removed. Thanks @QuintonJason

**Comment by ju-Skinner** [comment]
> This is looking great! Overall functionality is working as expected from what I'm seeing so far.
> 
> Question for the group: thoughts on keeping some aspect of the base "theme" structure in place? While `legacy`'s duplication should go, I could see a benefit in the ability to swap styles based off of CSS custom props/tokens, while keeping standard component markup/JS intact. But this being at the expense of complexity, I suppose a different approach to theming makes more sense.
> 
> There's also a few areas where the React/Storybook link is not functioning (Alert, Button, basically anywhere without a `Default` story view) but those are minor items that can be cleaned up later.

I thought about that as well. Would be a good use case for "theming", light / dark.

**Comment by philschanely** [comment]
@monkeypox8 no need for theming structure from my perspective. I believe if we find we need something along these lines we can devise a tactic for it specifically.

**Review by monkeypox8 — APPROVED** [review]
:shipit:

**Review by pixelflips — APPROVED** [review]
Clicked through docs and storybook and nothing jumped out. LGTM! 🔥 Great work!!

File: docs/app/views/examples/layouts/frame/_main.html.erb [hunk 2]
```diff
@@ -160,8 +160,8 @@ In such cases you can try `baseline` and often achieve a better result.
 |---|---|
 | `0px` | `none` |
 | `4px` | `sm` |
-| `#{SageRails.next_theme? ? "8px" : "6px"}` | `md` |
-| `#{SageRails.next_theme? ? "16px" : "8px"}` | `lg` |
+| `8px` | `md` |
+| `16p` | `lg` |
```
[the remark this answers is not in the document]
Comment (ju-Skinner) [reply docs/app/views/examples/layouts/frame/_main.html.erb:164]: Thanks @goodwinchris

**Review by goodwinchris — APPROVED** [review]
🌝

**Review by QuintonJason — APPROVED** [review]
Clicked through and it works great!

**Review by anechol — APPROVED** [review]

**Pull request merged**